### PR TITLE
Fixed the bookmark url matching when domain has www.

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1199525460093862/f
Tech Design URL: 
CC: 

**Description**:
Fixes the URL matching when returning bookmark results upon search query.

**Steps to test this PR**:
1. Install the app from this branch
2. Bookmark `https://www.reddit.com` and `https://www.reddit.com/r/duckduckgo`
3. In the search bar, search for `reddit`
4. Verify both bookmarks appear

(In develop branch, only `https://www.reddit.com` will appear)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt [hunk 1]
```diff
@@ -134,7 +134,7 @@ class AutoCompleteApi @Inject constructor(
                 rankedBookmark.score += 50
             }
 
-        } else if (bookmark.url.toUri().toStringDropScheme().startsWith(tokens.first().trimEnd { it == '/' }, ignoreCase = true)) {
+        } else if (bookmark.url.toUri().toStringDropScheme().removePrefix("www.").startsWith(tokens.first().trimEnd { it == '/' }, ignoreCase = true)) {
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt:137]: This logic is getting more difficult to read. Would you mind extracting it to a function that explains what it does for better readability?

**Review by marcosholgado — CHANGES_REQUESTED** [review]
Adding a suggestion to improve readability a bit for the else if.

File: app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt (hunk 1 above)
Comment (aitorvs) [reply app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt]: Done

**Review by marcosholgado — APPROVED** [review]
👍  LGTM!

**Pull request merged**